research-app/src/App.vue

```diff
@@ -1648,13 +1737,85 @@ export default class App extends WWTAwareComponent {
     return imagesets.filter(iset => iset.name.toLowerCase().includes(searchText) || iset.description.toLowerCase().includes(searchText));
   }
 
+  // RA, Dec in radians
+  distance(pt1: AngleCoordinates, pt2: AngleCoordinates): number {
+    // Using the last formula from https://en.wikipedia.org/wiki/Great-circle_distance#Computational_formulas
+    // which Wikipedia says is accurate at all distances
+    const dAbsRA = Math.abs(pt1.ra - pt2.ra);
+    const nt1 = (Math.cos(pt2.dec) * Math.sin(dAbsRA)) ** 2;
+    const nt2 = (Math.cos(pt1.dec) * Math.sin(pt2.dec) - Math.sin(pt1.dec) * Math.cos(pt2.dec) * Math.cos(dAbsRA)) ** 2;
+    const num = Math.sqrt(nt1 + nt2);
+    const den = Math.sin(pt1.dec) * Math.sin(pt2.dec) + Math.cos(pt1.dec) * Math.cos(pt2.dec) * Math.cos(dAbsRA);
+    return Math.atan2(num, den);
+  }
+
+  closestInView(target: AngleCoordinates, threshold?: number): Source | null {
+    let minDist = Infinity;
+    let closestPt = null;
+
+    //const pointsInView = this.dataInView();
+    //const result = this.closest(target, pointsInView);
+    // const closePt = result.pt
+    // const minDist = result.dist
+
+    const rowSeparator = "\r\n";
+    const colSeparator = "\t";
+
+    for (const catalog of this.visibleHipsCatalogs()) {
+      const name = catalog.name;
+      const hipsStr = this.currentHipsCatalogData(name);
+      const rows = hipsStr.split(rowSeparator);
+      const header = rows.shift();
+      if (!header) {
+        return null;
+      }
+      const colNames = header.split(colSeparator);
+
+      const lngCol = this.findLngColumn(name);
+      const latCol = this.findLatColumn(name);
+
+      const itemCreator = function (values: string[]): Source {
+        const obj: any = {};  // eslint-disable-line @typescript-eslint/no-explicit-any
+        for (let i = 0; i < values.length; i++) {
+          obj[colNames[i]] = values[i];
+        }
+        return { ...obj, ra: D2R * Number(values[lngCol]), dec: D2R * Number(values[latCol]), catalogName: name };
+      };
+
+      for (const row of rows) {
+        const items = row.split(colSeparator);
+        const ra = Number(items[lngCol]);
+        const dec = Number(items[latCol]);
+        const pt = { ra: D2R * ra, dec: D2R * dec };
+        const dist = this.distance(target, pt);
+        if (dist < minDist) {
+          closestPt = itemCreator(items);
+          minDist = dist;
+        }
+      }
+    }
+    if (!threshold || minDist < threshold) {
+      return closestPt;
+    }
+    return null;
+  }
+
   data() {
     return {
       showPopover: false,
       showLayers: true,
+      showSources: true,
     }
   }
 
+  updateDistanceThreshold = debounce((app: App, newZoom: number) => {
+    app.distanceThreshold = 0.00002 * newZoom;
```

Comment:
Where is this magic factor coming from? A quick explanatory comment would be helpful.

Reply:
I don't really have any reasoning for this factor - when I was testing out this functionality, I ended up landing on this value because I thought it 'felt good' as far as how close the cursor needed to get to a source before the cursor changed.

I was thinking - with the circle annotations that we're now going to have around the sources, maybe it would be better to use those distances as the threshold for each point?